src/Gui/Application.cpp

```diff
@@ -2175,6 +2175,23 @@ void Application::runApplication()
 
             int major = context.format().majorVersion();
             int minor = context.format().minorVersion();
+            if (major < 2) {
+                std::map<std::string, std::string>& config = App::Application::Config();
+                QString fcmajor  = QString::fromLatin1(config["BuildVersionMajor"].c_str());
+                QString fcminor  = QString::fromLatin1(config["BuildVersionMinor"].c_str());
+                QString fcpoint = QString::fromLatin1(config["BuildVersionPoint"].c_str());
+                QString msgboxtitle =
+                        QString::fromLatin1("%1 %2.%3.%4 ").arg(mainApp.applicationName(), fcmajor, fcminor, fcpoint);
+                QString bodyMessage;
+                QTextStream bodyMessageStream(&bodyMessage);
+                bodyMessageStream << QObject::tr("This system is running OpenGL version");
+                bodyMessageStream << " " << major << "." << minor;
+                bodyMessageStream << ", " << QObject::tr("FreeCAD requires OpenGL version");
+                bodyMessageStream << " 2.0 " << QObject::tr("or above");
```

Comment:
Does this structure work in all languages?

Reply:
It's possible to translate it into Russian, but the word order will be really weird.

Reply:
I've just realized that "or above" can be translated as "". It's possible to translate it normally, but in the end of the message there will be " 2.0 .".

Reply:
I think it might make more sense to just combine it into a single string, with the 2.0 embedded, or as a formatted string with a %f or whatever placeholder. Without context, translators are going to struggle to understand what is going on.